### ISSUE TITLE
Unable to display diff

### ISSUE THREAD

**Issue by Piszmog** [sourcegraph/sourcegraph#44953]

**Comment by courier-new** [comment]
Interesting! I don't have a good sense of when one would actually _want_ to programmatically modify binary files, so I don't know how important this is to address, or how to meaningfully address it.

**Comment by Piszmog** [comment]
I would agree that this is probably not super import. But I think it would be good to have a message around if this error occurs. 

When I ran this locally and it failed, I understood why. But when I ran in remote, at first I was shocked that it worked (green checks and everything). But then when I go to look at the results, I get this error. So it was not actually successful.

**Comment by chrispine** [comment]
IMO, doing _something_ about this is high or very-high for scale testing (even if that something is just a clearer error message)

**Comment by eseliger** [comment]
This seems like a regression - we supported binary diffs before and would simply say "cannot render binary files" in the UI - so something must've broken here. Since it's a bug, P0 or P1? 1d to investigate for now?

**Comment by eseliger** [comment]
So this happens to be an OOM issue on the server-side causing the 502. It does work for me locally but requires a few GB of RAM. Short term fix would be to just bump the memory of the scaletesting instance I guess.
The problem is that we load the oldfile content into memory (so the full ISO image). I'm not super sure how to fix it with a simple tweak somewhere, though, since we kinda need the file content for the IsBinary check and hypothetically highlighting after (which will abort after it's detected as binary). We might get away with a better FileDiff.IsBinary flag or so so we can early return, but for huge text files this issue would also exist.

The src-cli bug below looks to be unrelated and might be some bug in git or so if you google around a bit, since that's where the error is coming from. It somehow tries to allocate 16000 PB (!) to apply the patch lol. People on SO reported that a reboot fixed it for them but unclear what that means.

BTW the actual diff this generated is only 1.2MB in size.
